File: 250512/전부 겹치는 선분/overlapping-line-segments.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = parseInt(input[0]);
const segments = [];
for (let i = 1; i <= n; i++) {
    segments.push(input[i].split(' ').map(Number));
}
const visited = new Map()
for(let i = 0; i < segments.length; i++){
    const [x, y] = segments[i];
    for(let j = x; j <= y; j++){
        visited.set(j, (visited.get(j) || 0) + 1)
    }
}
let flag = false
for(val of visited.values()){
    if(val === n){
        flag = true
    }
}
console.log(flag ? 'Yes' : 'No')

File: 250818/전부 겹치는 선분/overlapping-line-segments.py
n = int(input())
segments = [tuple(map(int, input().split())) for _ in range(n)]
x1, x2 = zip(*segments)
x1, x2 = list(x1), list(x2)

flag = True

for i in range(1, n):
    a1, b1 = segments[i - 1]
    a2, b2 = segments[i]

    if b1 < a2 or b2 < a1:
        flag = False

if flag:
    print('Yes')
else:
    print('No')
